keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1132 @@
+  resource claim requests in pod's spec.resources.claims. Dynamic resource type
+  is described in resource slice, simply speaking, it is a list of devices, with
+  each device being described as structured parameters.
+* extended resource backend by DRA is a combination of the two above. It uses pods'
+  spec.containers[].resources.requests to request for resources, and uses
+  `ResourceSlice` to provide resource capacity. Hence, it is of type (string, int64)
+   on the consumption side, and list of devices with a common
+  `ExtendedResourceName` on the capacity side.
+
+With these additions in place, the DRA devices can be consumed by extended resource
+requests, or by DRA resouce claims. The scheduler has everything it needs to support
+the dynamic allocation of devices to requests made through extended resource and
+resource claims. No static partition of resources between extended resources and
+resource claims is needed. The kubelet and DRA driver has everything they need
+to admit a pod and pass the allocated devices to the containers in the pod to run.
+
+Note the following cluster setup configuration and constraint:
+
+* One node in a cluster can have an extended resource backed by DRA, and another node in the
+same cluster can have the same named extended resource backed by Device Plugin.
+
+* One node in a cluster cannot have both extended resource backed by DRA, and same
+  named extended resource backed by device plugin at the same time. This implies
+  that either the resource is advertised through `ResourceSlice`,
+  or `Node`'s status.capacity.
+
+## Design Details
+
+### Device Class API
+The extended resource name to DRA device mapping can be specified at
+`DeviceClassSpec`. The same extended resource name should be given to at most one 
+device class. If there are more than one device classes, the one created later is picked 
+at scheduling time, if two are created at the same time, the name
+lexicographically sorted first is picked.
+
+Cluster administrator is soly responsible for creating device classes, and the
+mapping between the class of devices and the extended resource name.
+`DeviceClass` is cluster scoped, application developers and operators cannot change it.
+
+The mapping of DRA devices and extended resources is stored in k8s data store
+(e.g. etcd). An application using the extended resources can only request the
+devices from DRA after the device class with the mapping is created. Before
+that, the application can request the devices from device plugin only.
+
+```go
+// DeviceClassSpec is used in a DeviceClass to define what can be allocated
+// and how to configure it.
+type DeviceClassSpec struct {
+	// ExtendedResourceName defines a mapping to the extended resource API.
+	// All devices matched by the device class can be used to satisfy extended resource requests in pod's spec using this name.
+	//
+	// +optional
+	ExtendedResourceName *string
```

Comment:
related to https://github.com/kubernetes/enhancements/pull/5136/files#r2136536134 , should we ensure the `ExtendedResourceName` matches the current device driver name to avoid problems fo dra drivers taken on other vendors resources?

Reply:
i think we can enforce the tld matches, .e.g requires both the extended resource (e.g. xyz.example.com/gpu) and driver (abc.example.com) have example.com

Reply:
that sounds a good idea that solves this problem

Reply:
Patrick, to be clear, I do not have a strong opinion and people here may have more use cases I'm not aware, but I feel that providing much flexibility to a mechanism that should be only solving the migration from device plugins to dra drivers, can make the scope creeping and we end in a situation like python2 -> python2 or ipv4->ipv6 that nobody nevers moves from the legacy mechanism

Reply:
resolving :)